src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -409,6 +436,17 @@ def __init__(
             self._schema_query = sql
             self._query_param = params
 
+    def __deepcopy__(self, memodict={}) -> "SelectSQL":  # noqa: B006
+        copied = SelectSQL(
```

Comment:
Will it trigger an extra describe call, when convert_to_select is true and the original sql is a non-select sql?

Reply:
That is a good point. Yes, it will when convert_to_select is True, updated the implementation to use false first, and then copy values over